src/clients/oracle.rs

```
@@ -475,6 +476,475 @@ impl OracleClient {
+                                    match result.extract::<bool>(py) {
+                                        Ok(decision) => Some(decision),
+                                        Err(_) => {
+                                            // If not a boolean, try to interpret as truthy/falsy
+                                            match result.is_truthy(py) {
+                                                Ok(truthy) => Some(truthy),
+                                                Err(_) => None,
+                                            }
+                                        }
+                                    }
+                                }
+                                Err(_) => None,
+                            }
+                        })
+                    };
+
+                // Create callback function that calls back to Python if provided
+                let callback = |decision: &alkahest_rs::clients::oracle::Decision<
+                    StringObligation::StatementData,
+                    (),
+                >| {
+                    // Always print the decision for debugging
+                    println!(
+                        "Decision made: {} for statement: {}",
+                        decision.decision, decision.statement.item
+                    );
+
+                    // Call Python callback if provided
+                    if let Some(ref py_callback) = callback_func {
+                        Python::with_gil(|py| {
+                            let decision_info = format!(
+                                "Decision: {} for statement: '{}'",
+                                decision.decision, decision.statement.item
+                            );
+
+                            // Call the Python callback function
+                            if let Err(e) = py_callback.call1(py, (decision_info,)) {
+                                println!("Error calling Python callback: {}", e);
```

can we bubble error instead of println?